packages/rum/src/apm-base.js

```diff
@@ -99,9 +99,33 @@ class ApmBase {
     return !this._disable
   }
 
+  /**
+   * When the required config keys are invalid, the agent is deactivated with
+   * logging error to the console
+   */
   config(config) {
-    var configService = this.serviceFactory.getService('ConfigService')
+    const configService = this.serviceFactory.getService('ConfigService')
     configService.setConfig(config)
+    /**
+     * Skip validation when agent is not active
+     */
+    if (!configService.isActive()) {
+      return
+    }
+
+    const errors = configService.validate()
+    if (errors.length === 0) {
+      return
+    }
+    const loggingService = this.serviceFactory.getService('LoggingService')
+    const message = 'RUM Agent configuration is invalid: ' + errors.join(', ')
+
+    /**
+     * TODO: Deactivate the agent when configuration is invalid
+     * on next major version
+     */
+    // configService.setConfig({ active: false })
```

Comment:
We can do this on next breaking change to align with other agents.

Reply:
We decided to make the change into a feature version.